crates/chain-state/src/in_memory.rs

```diff
@@ -819,7 +819,10 @@ pub struct ExecutedBlockWithTrieUpdates<N: NodePrimitives = EthPrimitives> {
     #[into]
     pub block: ExecutedBlock<N>,
     /// Trie updates that result of applying the block.
-    pub trie: Arc<TrieUpdates>,
+    ///
+    /// If [`None`], the trie updates should be computed when persisting the block **on top of the
+    /// canonical parent**.
+    pub trie: Option<Arc<TrieUpdates>>,
```

Comment:
i would prefer if we turned this into enum and added extensive docs describing when and why trie updates are missing
```rust
/// DOCS DOCS DOCS
enum ExecutedTrieUpdates {
  Present(Arc<TrieUpdates>),
  Missing,
}
```